experimental/LieAlgebras/src/LinearLieAlgebra.jl

```diff
@@ -317,37 +321,191 @@ function special_linear_lie_algebra(R::Field, n::Int)
   return L
 end
 
+function _lie_algebra_basis_from_form(R::Field, n::Int, form::MatElem)
+  invform = inv(form)
+  eqs = zero_matrix(R, n^2, n^2)
+  for i in 1:n, j in 1:n
+    x = zero_matrix(R, n, n)
+    x[i, j] = 1
+    eqs[(i-1) * n + j, :] = _vec(x + invform * transpose(x) * form)
+  end
+  ker = kernel(eqs)
+  rref!(ker) # we cannot assume anything about the kernel, but want to have a consistent output
+  dim = nrows(ker)
+  basis = [zero_matrix(R, n, n) for _ in 1:dim]
+  for i in 1:n
+    for k in 1:dim
+      basis[k][i, 1:n] = ker[k, (i-1) * n .+ (1:n)]
+    end
+  end
+  return basis
+end
+
 @doc raw"""
-    special_orthogonal_lie_algebra(R::Field, n::Int) -> LinearLieAlgebra{elem_type(R)}
+    special_orthogonal_lie_algebra(R::Field, n::Int, gram::MatElem) -> LinearLieAlgebra{elem_type(R)}
+    special_orthogonal_lie_algebra(R::Field, n::Int, gram::Matrix) -> LinearLieAlgebra{elem_type(R)}
 
 Return the special orthogonal Lie algebra $\mathfrak{so}_n(R)$.
 
+Given a non-degenerate symmetric bilinear form $f$ via its Gram matrix `gram`,
+$\mathfrak{so}_n(R)$ is the Lie algebra of all $n \times n$ matrices $x$ over the field `R`
+such that $f(xv, w) = -f(v, xw)$ for all $v, w \in R^n$.
+
+If `gram` is not provided, for $n = 2k$ the form defined by $\begin{matrix} 0 & I_k \\ -I_k & 0 \end{matrix}$
+is used, and for $n = 2k + 1$ the form defined by $\begin{matrix} 1 & 0 & 0 \\ 0 & 0 I_k \\ 0 & I_k & 0 \end{matrix}$.
```

Comment:
Why that form in particular? Is that the form you used before?

Reply:
No, I used the identity matrix before, which for general users is a bad choice, as eg over QQ the Cartan sub is not split.
This is the form used most often in the literature to construct the B_n and D_n types, eg in Humphreys Introduction to Lie Algebras. I think we should provide just any default here, and thus particular one has, in most cases, nice properties.